src/main/java/com/hida/Id.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package com.hida;
+
+import java.util.Arrays;
+
+/**
+ * An object used to model every Id. By definition, each Id will have a unique
+ * name associated with it. However, to determine uniqueness, each newly created
+ * Id must be compared to previously existing Ids.
+ *
+ * Comparisons will be made by using Sets collection. Depending on which set is
+ * used, the Comparable interface and an overridden equals and hashCode methods
+ * were overridden to accommodate.
+ *
+ * @author lruffin
+ */
+public abstract class Id implements Comparable<Id> {
+
+    // fields
+    private int[] BaseMap;
+    private boolean Unique = true;
+    private final String Prefix;
+
+    /**
+     * Copy constructor; primarily used to copy values of the BaseMap from one
+     * Id to another.
+     *
+     * @param id - The Id to copy from.
+     */
+    public Id(Id id) {
+        this.Prefix = id.Prefix;
+        this.BaseMap = Arrays.copyOf(id.getBaseMap(), id.getBaseMap().length);
+        this.Unique = id.Unique;
+    }
+
+    public Id(int[] baseMap, String PREFIX) {
+        this.BaseMap = Arrays.copyOf(baseMap, baseMap.length);
+        this.Prefix = PREFIX;
+    }
+
+    
+    public abstract boolean incrementId();
+
+    public abstract String convert();
+
+    
+    /**
+     * Generated by contract because equals was overridden. Prime numbers were
+     * chosen to lower the number of collisions. Increasing them to a higher
+     * prime number will lower collisions at the cost of run time.
+     *
+     * @return
+     */
+    @Override
+    public int hashCode() {
+        // arbitrarily chosen prime numbers
+        final int prime1 = 37;
+        final int prime2 = 257;
+
+        int hash = prime1 * prime2 + Arrays.hashCode(this.BaseMap);
+        return hash;
+    }
+
+    /**
+     * Overridden so that id's can be identified solely by its baseName.
+     *
+     * @param obj - the Object this id is being compared to
+     * @return - true if the two Objects are the same.
+     */
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return false;
+        }
+        if (!(obj instanceof Id)) {
+            return false;
+        }
+        final Id paramId = (Id) obj;
+
+        // equivalent to Arrays.hashCode(a) == Arrays.hashCode(b)

```

Comment:
Not necessarily.  Java only requires that if `equals()` is true, then the hash codes must match, but not the other way around.